app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -33,9 +39,10 @@ class FirebaseAuthRepository {
       context: Context,
       coroutineScope: CoroutineScope,
       credentialManager: CredentialManager = CredentialManager.create(context),
+      startAddAccountIntentLauncher: ManagedActivityResultLauncher<Intent, ActivityResult>?,
```

Comment:
Question: Why is the `startAddAccountIntentLauncher` nullable?

Reply:
The launcher is nullable since after connecting the account together was successful, we make the call to the authViewModel that I also show in my answer. 
```kotlin 
authViewModel.signInWithGoogle(coroutineScope, context, null)
```

If the launcher was not nullable, we would have to pass the addAccountLauncher itself as parameter here, which does not work afaik. It being null is not an issue however, since the NoCredentialError should be resolved after the google sign in is completed, so the addAccountLauncher, which is only ever used when a NoCredentialError is catched, should not need to be called.

Reply:
If the google sign in failed, this would result in the second call to the authViewModel to fail also, so ultimately just nothing will have happened. For example, if there is no internet, then the account connection will fail, and subsequently so will the second call to the authViewModel, and the user is just back to the sign in screen.

In that case the user can just click the sign in button again and retry.

Reply:
Thanks for the detailed explanation!